.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+name: Create new release from release branch
+
+# This action will only run when a pull request is closed/merged.
+on:
+  pull_request:
+    types:
+      - closed
```

Comment:
Should we add a `branches` filter to ensure this only runs on PRs that target `trunk`

e.g.

```
branches:
  - trunk
```

ref: https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-filters-to-target-specific-branches-or-tags-for-push-events

I think this will be the majority of our PRs but there might be circumstances where we branch off feature branches for certain projects.

Adding this filter would prevent this workflow from running on those branches when merged into the main feature branch.

Reply:
I find the documentation on this really confusing with `pull_request` vs `pull_request_target`. I've [tested](https://github.com/brucealdridge/gh_action_test/actions/runs/2093217595) and confirmed we should be using `pull_request_target` - the docs seem to indicate the reverse